src/index.ts

```diff
@@ -42,6 +42,7 @@ const call = async (request: Request, env: Env): Promise<any> => {
 		networks: [{
 			...networks[ChainId.POLYGON],
 			rpcUrl: nodeUrl,
+			provider: provider, // NOTE: must pass the provider here
```

Comment:
this was the key change to make it work... that is, we pass the specific StaticJsonRpcProvider from above which does the skipFetch stuff. Without this, the worker behaves strangely because ethers internally does a few things with timers / other strange things.